oauth_adapter.py

```diff
@@ -35,4 +39,47 @@ def refresh_access_token(self, refresh_token_str):
         """
         token_dict = {'refresh_token': refresh_token_str}
         oauth = OAuth2Session(self.client_id, token=token_dict)
-        return oauth.refresh_token(self.token_url, auth=self.basic_auth)
+        return oauth.refresh_token(self._get_token_info_url(), auth=self.basic_auth)
+
+    def revoke_refresh_token(self, refresh_token):
+        """
+        Calls the auth provider to revoke the given refresh token
+        :param refresh_token:
+        :return:
+        """
+        revoke_url = self._get_revoke_url()
+        result = urlfetch.fetch(url=revoke_url,
+                                method=urlfetch.POST,
+                                payload="token=" + refresh_token,
+                                headers={"Authorization": "Basic %s" % base64.b64encode("{}:{}".format(self.client_id, self.client_secret))})
+        if result.status_code // 100 != 2:
+            raise endpoints.InternalServerErrorException("revoke url {}, status code {}, error body {}".
+                                                         format(revoke_url, result.status_code, result.content))
+
+    def _get_open_id_config(self):
+        open_id_config = memcache.get(namespace="OauthAdapter", key=self.provider_name)
+        if not open_id_config:
+            open_id_config_response = urlfetch.fetch(self.open_id_config_url)
+            if open_id_config_response.status_code != 200:
+                raise endpoints.InternalServerErrorException(
+                    message='open_id_config_url [{}] returned status {}: {}'.format(self.open_id_config_url,
+                                                                                    open_id_config_response.status_code,
+                                                                                    open_id_config_response.content))
+            else:
+                open_id_config = json.loads(open_id_config_response.content)
+                memcache.add(namespace="OauthAdapter", key=self.provider_name, value=open_id_config)
```

Comment:
I know it's really unlikely to change, but just in case, should we put like a once-a-day expiry on this?

Reply:
I was thinking we could always just go into the google console and evict if we need to... but sure